### PR TITLE
OCPBUGS-33011: fix issues with Edit Route form

### PR DESCRIPTION
**Pull request by rhamilto**
After:

https://github.com/openshift/console/assets/895728/55c0f19f-24fe-49db-949f-7e777c39ab98

### CODE REVIEW

File: frontend/public/components/routes/create-route.tsx [hunk 1]
```diff
@@ -66,7 +66,7 @@ class CreateRouteWithTranslation extends React.Component<
     namespace: getActiveNamespace(),
     labels: {},
     portOptions: {},
-    alternateServices: [],
+    alternateBackends: [],
```
Comment (rhamilto) [inline frontend/public/components/routes/create-route.tsx:69]: Throughout the PR:  use `alternateBackends` to align with the Route schema.

File: frontend/public/components/routes/create-route.tsx [hunk 2]
```diff
@@ -181,45 +181,49 @@ class CreateRouteWithTranslation extends React.Component<
...
-      this.props.formik.setFieldValue('formData.alternateServices', updatedServiceEntriesArray);
-      if (updatedServiceEntriesArray.length < 1) {
-        this.setState({ weight: 100 });
-        this.props.formik.setFieldValue('formData.weight', 100);
-      }
-      return {
-        alternateServices: updatedServiceEntriesArray,
-      };
-    });
+    this.setState(
```
Comment (rhamilto) [inline frontend/public/components/routes/create-route.tsx:197]: Fixes a console error resulting from the nested `setState` calls.  Fix is to move the second `setState` to a callback.

File: frontend/public/components/routes/create-route.tsx [hunk 3]
```diff
@@ -342,7 +346,7 @@ class CreateRouteWithTranslation extends React.Component<
             className="pf-v5-c-form-control"
             type="text"
             onChange={this.handleChange}
-            value={this.state.path}
+            value={this.state.path ?? ''} // this.state.path can be set to undefined in componentDidMount
```
Comment (rhamilto) [inline frontend/public/components/routes/create-route.tsx:349]: Fixes a console error related to controlled component changing to an uncontrolled one.

File: frontend/public/components/routes/create-route.tsx [hunk 4]
```diff
@@ -426,7 +430,8 @@ class CreateRouteWithTranslation extends React.Component<
           {!_.isEmpty(portOptions) && (
             <Dropdown
               items={portOptions}
-              title={portOptions[targetPort] || t('public~Select target port')}
+              title={t('public~Select target port')}
+              selectedKey={_.isNumber(targetPort) ? UNNAMED_PORT_KEY : targetPort} // if targetPort is a number, it's an unnamed port
```
Comment (rhamilto) [inline frontend/public/components/routes/create-route.tsx:434]: Fix so the `Target port` dropdown is correctly populated when editing (before the dropdown did not have a selected value).

File: frontend/public/components/routes/utils.ts [hunk 5]
```diff
@@ -147,9 +147,7 @@ export const convertEditFormToRoute = (
     },
   };
 
-  if (!_.isEmpty(alternateBackends)) {
-    route.spec.alternateBackends = alternateBackends;
-  }
+  route.spec.alternateBackends = altBackends;
```
Comment (rhamilto) [inline frontend/public/components/routes/utils.ts:150]: This fixes the bug as reported in the Jira issue.  Passing an empty array is ok!

File: frontend/public/components/routes/RoutePage.tsx [hunk 6]
```diff
@@ -108,24 +109,33 @@ export const RoutePage: React.FC = () => {
...
 
   return (
-    <StatusBox loaded={loaded} loadError={routeLoadError || serviceLoadError} data={initialValues}>
-      <Formik
-        initialValues={initialValues}
-        onSubmit={handleSubmit}
-        validationSchema={routeValidationSchema}
-        enableReinitialize
+    <>
+      <Helmet>
```
Comment (rhamilto) [inline frontend/public/components/routes/RoutePage.tsx:113]: Add missing `<title>`

**Review by TheRealJon — APPROVED** [review]
/lgtm

**Pull request merged**